test/models/site_test.rb

```diff
@@ -13,22 +13,58 @@ class SiteTest < ActiveSupport::TestCase
   sub_test_case "relation" do
     setup do
       @site = create(:site)
-      @post = create(:post, :whatever, site: @site)
+      @post = create(:post, :whatever, :with_credit, site: @site)
       Tempfile.open do |file|
         @site.images.create!(image: file)
       end
+      create(:serial, site: @site)
+
+      create(:fixed_page, site: @site)
+      create(:link,       site: @site)
+      create(:membership, site: @site)
+      @site.redirect_rules.create!(request_path: "/foo", destination: "/bar")
```

Comment:
When I use the factory for `:redirect_rule` ( `create(:redirect_rule, :whatever, site: @site` ), this test is failed because of violation of a foreign key constraint. Are there any mistakes?